### PR TITLE
Cleaned up the code

### PR DESCRIPTION
**Pull request by maciekmm**
Cleaned it up and optimized a little.
Also you might look into your gravity/acceleration math because it's broken.

### CODE REVIEW

**Comment by pakastin** [comment]
Thanks, I'll check this out! Did you notice I just fixed collisions..?

**Comment by maciekmm** [comment]
@pakastin Yes, I've merged it already. But acceleration is still kinda broken, look into calculating it using trigonometry.

**Comment by pakastin** [comment]
Cool, you're fast! ;)
hmm.. sorry it's hard to follow what's changed because so much refactoring.. What was the issue(s)?

**Comment by pakastin** [comment]
Which part of the gravitation math needs fixing?

**Comment by maciekmm** [comment]
The issue is that the code doesn't follow almost any standards.
- I've added squared distance check as rooting is quite heavy.
- I've converted it to ES6.
- And cleaned it all up.
Merge it locally and check what has changed :).


https://github.com/pakastin/nodegarden/blob/master/scripts/index.js#L149-L159

**Comment by pakastin** [comment]
Sorry, I still don't understand how your lines differ:
```
var xAccA = force * direction.x / nodeA.mass;
var xAccB = force * direction.x / nodeA.mass;
var yAccA = force * direction.y / nodeB.mass;
var yAccB = force * direction.y / nodeB.mass;

// calculate new velocity after gravity
nodeA.velocityX += xAccA;
nodeA.velocityY += yAccA;

nodeB.velocityX -= xAccB;
nodeB.velocityY -= yAccB;
```

About those standards: I use [standard](https://github.com/feross/standard) code style and don't like ES6 classes / fat arrows.. I agree with the mess though – will refactor it my way, but merge to this pull request. Thank you so much for contribution!

**Comment by maciekmm** [comment]
@pakastin They don't. I've not fixed it. I can downgrade to ES5. But babel does this fine. ES6 is just around the corner for browsers and provides really readable syntax.

**Comment by pakastin** [comment]
I just don't like the ugly code babel makes with classes and fat arrows..but you have great things going on – I'll show you my fixes in a moment..

**Comment by zeroZshadow** [comment]
What he means is that xAccB is using nodeA's mass, instead of B's mass.
Also the temporary variabled are only used once, thus totally useless. (except of maby documentation)

**Comment by maciekmm** [comment]
@pakastin Should I merge your new commits to this PR?

**Comment by pakastin** [comment]
I can fix this, no worries ;) Thank you so much – works great now!

**Comment by pakastin** [comment]
Thanks for contribution!

**Comment by maciekmm** [comment]
You have not implemented squareddistance check which i think is a great performance improvement. 
I also find setting fill color and doing night check on every render call strange, same with variable declarations at the beginning of the function.

I'll contribute tomorrow by implementing #1 and some nicer screen edge behavior.

**Comment by pakastin** [comment]
I'm happy to merge, sorry I made this refactoring straight to the master branch, but there would've been something like 99.9% undoing and too much of a hassle pulling/pushing. Hope you're ok with my code style.

Didn't have time to investigate the square root optimization any further. I did realize there's more squaring than square rooting going on, so I believe it would make a difference. If you want you can make pull request for it.

I didn't notice your fix on setting fill and night check, sounds good also – I let that one for you to pull request as well if you want. Don't want to "steal" your contributions any more ;)
